Code/Editor/Util/Image.cpp

```diff
@@ -75,17 +75,14 @@ void CImageEx::ReverseUpDown()
     }
 
     uint32* pPixData = GetData();
-    uint32* pReversePix = new uint32[GetWidth() * GetHeight()];
 
-    for (int i = GetHeight() - 1, i2 = 0; i >= 0; i--, i2++)
+    for (int i = 0; i < GetHeight()/2; i++)
     {
         for (int k = 0; k < GetWidth(); k++)
         {
-            pReversePix[i2 * GetWidth() + k] = pPixData[i * GetWidth() + k];
+            AZStd::swap(pPixData[(GetHeight() - i) * GetWidth() + k], pPixData[i * GetWidth() + k]);
         }
     }
```

Comment:
your change looks good
Some minor recommendations (remove the comments if you take it):
```
const int halfHeight = GetHeight() / 2; // << precompute 
for (int i = 0; i < halfHeight ; ++i) // << cosmetic, pre-increment preferred
{
    const int columnOffset = i * GetWidth(); // << precompute 
    const int inverseColumnOffset = (GetHeight() - i) * GetWidth(); // << precompute 
    for (int k = 0; k < GetWidth(); ++k)
    {
        AZStd::swap(pPixData[inverseColumnOffset + k], pPixData[columnOffset + k]);
    }
}
```
You could also try to optimize further by using 64-bit operations, but you will have to handle the case where you have odd number of columns

Reply:
The big win here is only doing half the work. The rest are micro optimizations that will provide further improvements such as using larger steps using 64-bit values or using SIMD, loop unrolling, memcpy-ing. Another improvement would be to avoid refetching values e.g. by doing the following:
```
int halfHeight = GetHeight() / 2;
int width = GetWidth();
int doubleWidth = GetWidth() * 2;

uint32* frontLine = pPixData;
uint32* backLine = pPixData + (GetHeight() - 1) * width;

for (int h = 0; h < halfHeight ; ++h)
{
    for (int w = 0; w < width; ++w)
    {
        AZStd::swap(*backLine, *frontLine);
        frontLine++;
        backLine++;
    }
    backLine -= doubleWidth;
}
```
(This is drycoded so ymmv.)